Added Chat history similar to Grok, with Chatpreview panel on Right side

Finityno opened the pull request:
Have created a Chat preview chat history similar to grok,

i have added a const MIN_LOADING_TIME_MS = 300;

This could be lowered to preference, the action bar could be improved, and looks a bit ugly right now, but might be a starting point for a new chat history similar to grok, to improve UX

ibelick commented:
hey @Finityno, very cool, thank you for the work!
some first notes (didn’t check the code yet):

- can we remove the “actions”? <img width="366" alt="Capture d’écran 2025-04-24 à 14 09 47" src="https://github.com/user-attachments/assets/a1291793-bb8a-4c4e-829e-b73eced8fcf5" />

- I can’t delete chats / don’t see the delete button <img width="362" alt="Capture d’écran 2025-04-24 à 14 09 39" src="https://github.com/user-attachments/assets/e31e778f-6f27-4053-a2c9-42d131811084" />

- can we keep top/down keystrokes to navigate chats?

thank you

Finityno commented:
Yes, i will look into this, i just quickly ported my own version, that unfortunately was using a paid ui library, so ported it over quickly, mostly made this for fun(on both versions)

Might see if i find a way to match the version i have more closely, at least fix intended features for now.

![image](https://github.com/user-attachments/assets/37c8670c-52ba-49ff-82de-2f7686cdd8db)

Finityno commented:
Actions are supposed to work like this, and only show if you are in an excisting chat, and hid when not in a chat, similar to Grok, but not the full action bar, but might be unnecessary of course, but just a nice flow

![image](https://github.com/user-attachments/assets/6c4298c8-f2d4-4ce1-bc41-556bf85a4926)

ibelick commented:
hey @Finityno, can you rebase please? I updated the dialog history a bit yesterday. Once that’s done I’ll test your work, thank you!

Finityno commented:
Sure,

Do you want me to add and keep the new KBD stuff you added, and havent quite looked at the other changes, but from what i quickly saw, add aria labeling support etc as well?

ibelick commented:
@Finityno yes that would be great!
something nice to consider (maybe not for this PR) would be adding a  "show conversation preview" bottom left button, like  the one in Grok. I think it would improve the dialog too.

Finityno commented:
Like this tooltip(1 image is grok)

I have added support for showing with and without chat preview, in the button in the bottom left corner, but could definetely make it clearer in the button, i have a tooltip though as seen in the third image below

But the animation from going from the different modes, are a bit clunky at the moment,

Will take a look at this later today thanks.

![image](https://github.com/user-attachments/assets/f0a149f9-7a29-48d0-8fe7-e4a456d0fd9f)
![image](https://github.com/user-attachments/assets/71badb90-2412-4845-88f5-4d2723f1ed6d)
![image](https://github.com/user-attachments/assets/c367ff31-f358-45fc-b916-3b613e06753a)

Finityno commented:
Added support for math rendering, now as well, but probably should add that in a separate PR without command history, let me know

![image](https://github.com/user-attachments/assets/f74a35e7-2f31-4685-811e-c4f6542e35c2)

Also the command history currently looks like this with the KBD footer you added in it now, and reflecting the new changes made to dialog in terms of color and removed borders in command.tsx etc etc,
With Chatpreview
![image](https://github.com/user-attachments/assets/7662cb75-db52-475c-96c7-a64c5f363e70)

Without Chatpreview
![image](https://github.com/user-attachments/assets/25aebb5c-b248-4f1f-8160-cd40b7cb1bed)

ibelick commented:
hey @Finityno, is it ready for review?
Could you move all the math rendering code into a separate PR?
I think it'll need a bit more work since the system prompt can be rewritten per agent.
I was thinking of adding an "additional system prompt" feature soon, with like the current date and user preference (in the future like locales / languages). The LaTeX math inscrution could go there too.

thank you for the work!

Finityno commented:
Hi,

If you we're able to quickly go over the code, like not in detail, but the way the different components are set up, And suggest how you want the command history stuff setup, like in how many components for it in total, that would be appreciated, and then i could look further into the code,

I could just consider adding it in app/components/history/ and a separate ui folder for all the components for command item rows etc etc, but i think i can merge multiple of those components into the same utils file or something like that

I have checked the component (command history)with react scan, and the performance looks good, 

Also btw real quick, i cant see the history button in the header, when i have selected the command history appearance, can add this to issues.

I can move the math rendering to a separate PR; but it was really easy to implement, so you can consider checking out my code and just copying it, or just setting up again yourself if thats easier for you, and you can set it up in your way, but i can definetely set up a pr for it if you want that as well,

I did also add in this pr a way to make sure the sidebar, when scrolling down and going on a new chat, the scroll bar stay consistent, can add this as a separate PR if you want that as well

![image](https://github.com/user-attachments/assets/eba0788a-662c-42e7-8f85-869e65d75546)

ibelick commented:
hey @Finityno
would be super helpful if you could split this into separate PRs: 
- one for the command history changes (and scrollbar persistence)
- one for math rendering
it’ll make the review much easier for me

thanks again!
